packages/database/src/tests/cachers/EntityHierarchyCacher.fixtures.js

```diff
@@ -74,44 +72,350 @@ export const TEST_DATA = {
+// generation
+export const HIERARCHY_B_AFTER_PARENT_ID_CHANGES = expandEntityRelations([
+  // ancestor entity a
+  ['a', 'aa', 1],
+  ['a', 'ab', 2],
+  ['a', 'aba', 3],
+  ['a', 'aaa', 4],
+  ['a', 'abb', 5],
+  ['a', 'aab', 6],
+  // ancestor entity aa
+  ['aa', 'ab', 1],
+  ['aa', 'aba', 2],
+  ['aa', 'aaa', 3],
+  ['aa', 'abb', 4],
+  ['aa', 'aab', 5],
+  // ancestor entity ab
+  ['ab', 'aba', 1],
+  ['ab', 'aaa', 2],
+  ['ab', 'abb', 3],
+  ['ab', 'aab', 4],
+  // ancestor entity aba
+  ['aba', 'aaa', 1],
+  ['aba', 'abb', 2],
+  ['aba', 'aab', 3],
+  // ancestor entity aaa
+  ['aaa', 'abb', 1],
+  ['aaa', 'aab', 2],
+  // ancestor entity abb
+  ['abb', 'aab', 1],
+]);
+
+// test moves aab to below aa, and ab to below aab, resulting in:
+//      a
+//      aa
+//      aab
+//      ab
+//   aba  abb
+//   aaa
+export const HIERARCHY_B_AFTER_MULTIPLE_RELATIONS_CHANGED = expandEntityRelations([
+  // ancestor entity a
+  ['a', 'aa', 1],
+  ['a', 'aab', 2],
+  ['a', 'ab', 3],
+  ['a', 'aba', 4],
+  ['a', 'abb', 4],
+  ['a', 'aaa', 5],
+  // ancestor entity aa
+  ['aa', 'aab', 1],
+  ['aa', 'ab', 2],
+  ['aa', 'aba', 3],
+  ['aa', 'abb', 3],
+  ['aa', 'aaa', 4],
+  // ancestor entity aab
+  ['aab', 'ab', 1],
+  ['aab', 'aba', 2],
+  ['aab', 'abb', 2],
+  ['aab', 'aaa', 3],
+  // ancestor entity ab
+  ['ab', 'aba', 1],
+  ['ab', 'abb', 1],
+  ['ab', 'aaa', 2],
+  // ancestor entity aba
+  ['aba', 'aaa', 1],
+]);
+
+// adding a new entity relation record between aba and ab means that the canonical links will no
+// longer be used, so it will drop the abb subtree
+// a
+// aa
+// ab
+// aba
+// aaa
+export const HIERARCHY_B_AFTER_ENTITY_RELATION_ADDED = expandEntityRelations([
+  // ancestor entity a
+  ['a', 'aa', 1],
+  ['a', 'ab', 2],
+  ['a', 'aba', 3],
+  ['a', 'aaa', 4],
+  // ancestor entity aa
+  ['aa', 'ab', 1],
+  ['aa', 'aba', 2],
+  ['aa', 'aaa', 3],
+  // ancestor entity ab
+  ['ab', 'aba', 1],
+  ['ab', 'aaa', 2],
+  // ancestor entity aba
+  ['aba', 'aaa', 1],
+]);
+
+// entity abc is created below ab, entity aaaa below aaa, and ac below a
+//                 a
+//    aa           ab           ac
+// aaa  aab   aba  abb  abc     aca
+// aaaa
+export const HIERARCHY_A_AFTER_ENTITIES_CREATED = expandEntityRelations([
+  // ancestor entity a
+  ['a', 'aa', 1],
+  ['a', 'ab', 1],
+  ['a', 'ac', 1],
+  ['a', 'aaa', 2],
+  ['a', 'aab', 2],
+  ['a', 'aba', 2],
+  ['a', 'abb', 2],
+  ['a', 'abc', 2],
+  ['a', 'aca', 2],
+  ['a', 'aaaa', 3],
+  // ancestor entity aa
+  ['aa', 'aaa', 1],
+  ['aa', 'aab', 1],
+  ['aa', 'aaaa', 2],
+  // ancestor entity ab
+  ['ab', 'aba', 1],
+  ['ab', 'abb', 1],
+  ['ab', 'abc', 1],
+  // ancestor entity ac
+  ['ac', 'aca', 1],
+  // ancestor entity aaa
+  ['aaa', 'aaaa', 1],
+]);
+
+// entity abc is created below ab, entity aaaa below aaa, and ac below a, and aca below the new ac
+// in hierarchy b, the ac subtree is ignored as entity relation links are used at that generation
```

Comment:
Again, on a second read "entity relation links" are accurate here, perhaps speaking in terms of "canonical" and "alternate" hierarchies may make this a bit easier to understand

Reply:
True. I've been leaning away from the word 'alternate' or 'alternative' hierarchies, because I feel it leads people to treat them as the exceptional case, whereas I'd like us to lean more into them and away from the canonical ones over time. However, if it's causing confusion, it is better to be clear! I'll reword this a bit.